ktor-client/ktor-client-cio/jvm/src/io/ktor/client/engine/cio/ConnectionFactory.kt

```diff
@@ -4,23 +4,28 @@
 
 package io.ktor.client.engine.cio
 
-import io.ktor.util.cio.*
 import io.ktor.network.selector.*
 import io.ktor.network.sockets.*
 import io.ktor.network.sockets.Socket
+import io.ktor.network.sockets.SocketOptions
 import kotlinx.coroutines.sync.*
 import java.net.*
+import kotlin.coroutines.*
 
 internal class ConnectionFactory(
     private val selector: SelectorManager,
     maxConnectionsCount: Int
 ) {
     private val semaphore = Semaphore(maxConnectionsCount)
 
-    suspend fun connect(address: InetSocketAddress): Socket {
+    suspend fun connect(
+        address: InetSocketAddress,
+        parentContext: CoroutineContext = EmptyCoroutineContext,
+        configure: SocketOptions.TCPClientSocketOptions.() -> Unit = {}
+    ): Socket {
         semaphore.acquire()
         return try {
-            aSocket(selector).tcpNoDelay().tcp().connect(address)
+            aSocket(selector).tcpNoDelay().tcp().connect(address, parentContext, configure)
```

Comment:
Parent redundant here

Reply:
Do you mean naming? I renamed it to just `context`.